sound/soc/soc-compress.c

```diff
@@ -115,6 +126,14 @@ static int soc_compr_open(struct snd_compr_stream *cstream)
 		cpu_dai->driver->cops->shutdown(cstream, cpu_dai);
 out:
 	mutex_unlock(&rtd->card->pcm_mutex);
+
+	for_each_rtd_components(rtd, rtdcom, component) {
+		if (component == save)
+			continue;
```

Comment:
How about doing a break here?

We are in the open stage, so the card is fully initialized and one can assume that the list of components is fixed.
So if you have an error, you bail and mark all previously processed components as last busy/put_autosuspend.

No risk of leaks when going past the initial error

Reply:
Aligned with your suggestion in v2.